joomla/joomla-cms
# [4.3] Do not track rows in #__ucm_content as assets

Pull Request for Issue #37153.

### Summary of Changes
The UCM system currently spams the assets table with useless entries. Each row in the ucm_content table gets a corresponding row in the assets table which is a direct child of the root node and doesn't have any children itself. Their content always is empty, since the assets data is not copied over from the original content and also is never used anywhere. This PR stops the ucm_content table class from tracking these rows as assets and deletes the existing records in the assets table.

The delete query will not break the tree, since all these entries are leaf nodes, so it will not leave behind any orphaned children. The only inconsistencies by this are holes in the lft and rgt values, however the nested sets model is resilient against these holes and thus this shouldn't create an issue. To close these holes in these values, we would have to load the assets table class and call a `$table->rebuild()` on this, but since this is a time-consuming task and especially for large sites will most likely generate timeouts, I'm leaving this out here.

Doing stuff like such a rebuild would be something for a maintenance component in the future.

### Testing Instructions
1. Create an article and save it
2. Go to #__assets and see the new entry for your article. See directly below another new entry with a name and title similar to `#__ucm_content.42` and the rules column having the value `{}`.
3. Apply patch and create another article.
4. See that there is no new `#__ucm_content.69` entry in the assets table accompanying your articles asset entry.
5. Execute the included query to clear your assets table from all the dummy entries.
6. See that everything works as before.

### Link to documentations
Please select:
